tests/network/localnet/conftest.py

```diff
@@ -38,81 +57,185 @@ def nncp_localnet() -> Generator[libnncp.NodeNetworkConfigurationPolicy]:
         yield nncp
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def namespace_localnet_1() -> Generator[Namespace]:
     yield from create_ns(name="test-localnet-ns1")  # type: ignore
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def namespace_localnet_2() -> Generator[Namespace]:
     yield from create_ns(name="test-localnet-ns2")  # type: ignore
 
 
-@pytest.fixture(scope="module")
-def vlan_id(vlan_index_number: Generator[int]) -> int:
-    return next(vlan_index_number)
-
-
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def nad_localnet_1(
     namespace_localnet_1: Namespace, vlan_id: int
 ) -> Generator[netattachdef.NetworkAttachmentDefinition]:
-    with localnet_nad(namespace=namespace_localnet_1.name, name="test-localnet-nad1", vlan_id=vlan_id) as nad:
+    with localnet_nad(
+        namespace=namespace_localnet_1.name, name="test-localnet-nad1", vlan_id=vlan_id, network_name=BR_EX_NETWORK_NAME
+    ) as nad:
         yield nad
 
 
-@pytest.fixture(scope="module")
+@pytest.fixture(scope="class")
 def nad_localnet_2(
-    namespace_localnet_2: Namespace, vlan_id: int
+    nncp_localnet: Generator[libnncp.NodeNetworkConfigurationPolicy], namespace_localnet_2: Namespace, vlan_id: int
```

Comment:
`nncp_localnet` is ref but its value not used. Better ref it with a decorator marker.